src/Domain/CoordinateMaps/BulgedCube.hpp

```diff
@@ -0,0 +1,506 @@
+ *
+ *  ### The Spherical Face Map
+ *  The surface map for the spherical face of radius \f$R\f$ lying in the
+ *  \f$+z\f$
+ *  direction in either choice of coordinates is then given by:
+ *
+ *  \f[\vec{\sigma}_{spherical}: \vec{\xi} \rightarrow \vec{x}(\vec{\xi})\f]
+ *  Where
+ *  \f[
+ *  \vec{x}(\xi,\eta) =
+ *  \begin{bmatrix}
+ *  x(\xi,\eta)\\
+ *  y(\xi,\eta)\\
+ *  z(\xi,\eta)\\
+ *  \end{bmatrix}  = R
+ *  \begin{bmatrix}
+ *  \Xi/\rho\\
+ *  \mathrm{H}/\rho\\
+ *  1/\rho\\
+ *  \end{bmatrix}
+ *  \f]
+ *
+ *  ### The Cubical Face Map
+ *  The surface map for the cubical face of side length \f$2L\f$ lying in the
+ *  \f$+z\f$ direction is given by:
+ *
+ *  \f[\vec{\sigma}_{cubical}: \vec{\xi} \rightarrow \vec{x}(\vec{\xi})\f]
+ *  Where
+ *  \f[
+ *  \vec{x}(\xi,\eta) =
+ *  \begin{bmatrix}
+ *  x(\xi,\eta)\\
+ *  y(\xi,\eta)\\
+ *  L\\
+ *  \end{bmatrix}  = L
+ *  \begin{bmatrix}
+ *  \Xi\\
+ *  \mathrm{H}\\
+ *  1\\
+ *  \end{bmatrix}
+ *  \f]
+ *
+ *  ### The Bulged Face Map
+ *  To construct the bulged map we interpolate between this cubical face map
+ *  and a spherical face map of radius `radius_of_other_surface`, with the
+ *  interpolation parameter being \f$s\f$, the `sphericity`.
+ *  The surface map for the bulged face lying in the \f$+z\f$ direction is then
+ *  given by:
+ *
+ *  \f[
+ *  \vec{\sigma}_{+z}(\xi,\eta) = \left\{(1-s)L + \frac{sR}{\rho}\right\}
+ *  \begin{bmatrix}
+ *  \Xi\\
+ *  \mathrm{H}\\
+ *  1\\
+ *  \end{bmatrix}
+ *  \f]
+ *
+ *  We constrain L by demanding that the spherical face circumscribe the cube.
+ *  With this condition, we have \f$L = R/\sqrt3\f$.
+ *
+ *  ### The General Formula for 3D Isoparametric Maps
+ *  The general formula is given by Eq. 1 in section 2.1 of Hesthaven's paper
+ *  "A Stable Penalty Method For The Compressible Navier-Stokes Equations III.
+ *  Multidimensional Domain Decomposition Schemes" available
+ *  <a href="
+ *  http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.699.1161&rep=rep1&type=pdf
+ *  "> here </a>.
+ *
+ *  Hesthaven's formula is general in the degree of the shape functions used,
+ *  so for our purposes we take the special case where the shape functions are
+ *  linear, and define new variables accordingly.
+ *
+ *  We define the following variables for
+ *  \f$\alpha, \beta, \gamma \in\{\xi,\eta,\zeta\}\f$:
+ *
+ *  \f[
+ *  f^{\pm}_{\alpha} = \frac{1}{2}(1\pm\alpha)\\
+ *  f^{\pm\pm}_{\alpha \ \beta} = \frac{1}{4}(1\pm\alpha)(1\pm\beta)\\
+ *  f^{\pm\pm\pm}_{\alpha \ \beta \ \gamma} =
+ *  \frac{1}{8}(1\pm\alpha)(1\pm\beta)(1\pm\gamma)
+ *  \f]
+ *
+ *  The formula involves six surfaces, which we will denote by
+ *  \f$\vec{\sigma}\f$, twelve curves, denoted by \f$\vec{\Gamma}\f$, and eight
+ *  vertices, denoted by \f$\vec{\pi}\f$, with the subscripts denoting which
+ *  face(s) these objects belong to. The full volume map is given by:
+ *
+ *  \f{align}
+ *  \phi(\xi,\eta,\zeta) =
```

Comment:
change trailing operators to leading, 
\f{align*}
 + *  \phi(\xi,\eta,\zeta) =&
...
&f^{+}_{\ in the case where this is after a "\\\\" new line
...